public/components/ClearCacheModal/ClearCacheModal.tsx

```diff
@@ -0,0 +1,181 @@
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiSpacer,
+} from "@elastic/eui";
+
+interface ClearCacheModalProps<T> {
+  selectedItems: T[];
+  visible: boolean;
+  onClose: () => void;
+  type: string;
+}
+
+export default function ClearCacheModal<T>(props: ClearCacheModalProps<T>) {
+  const { onClose, visible, selectedItems, type } = props;
+  const [hint, setHint] = useState("");
+  const [blockHint, setBlockHint] = useState("");
+  const [unBlockedItems, setUnBlockedItems] = useState([] as string[]);
+  const [blockedItems, setBlockedItems] = useState([] as string[]);
+
+  const services = useContext(ServicesContext);
+  const coreServices = useContext(CoreServicesContext) as CoreStart;
+
+  useEffect(() => {
+    const indexBlocksTypes = [
+      IndexOpBlocksType.Closed,
+      IndexOpBlocksType.MetaData,
+      IndexOpBlocksType.ReadOnly,
+      IndexOpBlocksType.ReadOnlyAllowDelete,
+    ];
+    if (!!services && !!visible) {
+      switch (type) {
+        case "data streams":
+          setHint("Caches of the backing indexes behind the following data streams will be cleared.");
+          setBlockHint(
+            "Caches of the backing indexes behind the following data streams will not be cleared because of index closed or other blocks."
+          );
+          filterBlockedItems<DataStream>(services, selectedItems as DataStream[], indexBlocksTypes, dataStreamBlockedPredicate).then(
+            (filteredDataStreamsResult) => {
+              setUnBlockedItems(filteredDataStreamsResult.unBlockedItems.map((item) => item.name));
+              setBlockedItems(filteredDataStreamsResult.blockedItems.map((item) => item.name));
+            }
+          );
+          break;
+        case "aliases":
+          setHint("Caches of the indexes behind the following aliases will be cleared.");
+          setBlockHint("Caches of the indexes behind the following aliases will not be cleared because of index closed or other blocks.");
+          filterBlockedItems<IAlias>(services, selectedItems as IAlias[], indexBlocksTypes, aliasBlockedPredicate).then(
+            (filteredAliasesResult) => {
+              setUnBlockedItems(filteredAliasesResult.unBlockedItems.map((item) => item.alias));
+              setBlockedItems(filteredAliasesResult.blockedItems.map((item) => item.alias));
+            }
+          );
+          break;
+        default:
+          setHint("Caches of the folowing indexes will be cleared.");
+          setBlockHint("Caches of the following indexes will not be cleared because of index closed or other blocks.");
+          filterBlockedItems<CatIndex>(services, selectedItems as CatIndex[], indexBlocksTypes, indexBlockedPredicate).then(
+            (filteredIndexesResult) => {
+              setUnBlockedItems(filteredIndexesResult.unBlockedItems.map((item) => item.index));
+              setBlockedItems(filteredIndexesResult.blockedItems.map((item) => item.index));
+            }
+          );
+      }
+    }
```

Comment:
It would be better that we reset the state of blockItems/unBlockItems to empty array when the visible state changes to false.